packages/flutter/lib/src/services/raw_keyboard.dart

```diff
@@ -494,20 +504,50 @@ class RawKeyboard {
 
   final List<ValueChanged<RawKeyEvent>> _listeners = <ValueChanged<RawKeyEvent>>[];
 
-  /// Calls the listener every time the user presses or releases a key.
+  /// Register a listener that is called every time the user presses or releases
+  /// a hardware keyboard key.
+  ///
+  /// Since they have no way to indicate what they did with the event, listeners
+  /// are assumed to not handle the key event, and so these events will also be
+  /// distributed to other listeners, and to the [keyEventHandler].
+  ///
+  /// Most applications can use the focus system (see [Focus] and
+  /// [FocusManager]) to receive key events instead of this kind of passive
+  /// listener.
   ///
   /// Listeners can be removed with [removeListener].
   void addListener(ValueChanged<RawKeyEvent> listener) {
     _listeners.add(listener);
   }
 
-  /// Stop calling the listener every time the user presses or releases a key.
+  /// Stop calling the given listener every time the user presses or releases a
+  /// key.
   ///
   /// Listeners can be added with [addListener].
   void removeListener(ValueChanged<RawKeyEvent> listener) {
     _listeners.remove(listener);
   }
 
+  /// A handler for key events that will stop propagation if it returns true.
+  ///
+  /// This handler is normally set by the [FocusManager] so that it can control
+  /// the key event propagation to focused widgets.
+  ///
+  /// Most applications can use the focus system (see [Focus] and
+  /// [FocusManager]) to receive key events. If you are not using the
+  /// [FocusManager] to manage focus, then to be able to stop propagation of the
+  /// event, and indicate that the event was handled, set this attribute to a
+  /// [RawKeyEventHandler]. Otherwise, key events will be assumed to not have
+  /// been handled, and will also be sent to other (possibly non-Flutter)
+  /// controls in the application.
+  ///
+  /// See also:
+  ///
+  ///  * [Focus.onKey], a [Focus] callback attribute that will be given key
```

Comment:
maybe link to [addListener] as well from here and say how that's different then receiving the key events from here.